core/sql/langman/LmRoutineJavaObj.cpp

```diff
@@ -192,6 +192,7 @@ LmResult LmRoutineJavaObj::invokeRoutineMethod(
           (*emitRowPtr_)(NULL,0,&qs);
         }
     }
+  jni->DeleteLocalRef(jniResult);
```

Comment:
That can cause problems too. If there are no references retained, it can be gc-ed by JVM leading to unknown results when these variables are accessed. Is this reason this class was made static.  If so, there is another issue that this is not thread safe.